### ISSUE TITLE
MailPoet overrides From/Reply-To settings for non-MailPoet emails

### ISSUE THREAD

**Issue by ericnicolaas** [mailpoet/mailpoet#2799]
**Describe the bug**
My plugin ([Charitable](https://wordpress.org/plugins/charitable/)) sends donation receipts and other emails, and it allows users to specify From Name & Email Address settings. These are then used when sending, via `wp_mail()`. 

However, when using MailPoet with the "Send all site’s emails with…" setting set to use MailPoet's sending method, the From Name & Email Address are replaced with MailPoet's "Default sender" settings.

**To reproduce**
1. Install Charitable & MailPoet.
2. Set up Charitable: Create campaign, activate Offline payment gateway (Charitable > Settings > Payment Gateways), enable Donation Receipt email (Charitable > Settings > Emails), set "From" Name and "From" Email settings to something different than MailPoet settings.
3. Make a donation.

The donation receipt you receive will be using MailPoet's settings, not the ones configured in Charitable.

**Actual behavior**
Emails are sent with the "Default sender" settings added by MailPoet, not the ones provided by Charitable.

**Expected behavior**
The emails should be sent with the ones provided by Charitable, instead of MailPoet overriding them.

**Versions (please complete the following information):**
- WordPress version: 5.4.1
- PHP version: 7.2.9
- MailPoet version: 3.46.8
- Charitable version: 1.6.39

**Comment by ericnicolaas** [comment]
Just to add: I understand this is a plugin conflict and I'm prepared to fix this within Charitable, but I suspect that this is likely to be an issue for MailPoet with other plugins too — any plugin that sends transactional emails and allows you to define the sender details.

**Comment by pavel-mailpoet** [comment]
Hello ericnicolaas, 

thank you for your report. But I am afraid we will not fix this. What you described is the expected behaviour. When emails are sent through MailPoet, the plugin needs to update the from address.

Pavel

**Comment by ericnicolaas** [comment]
Hi @pavel-mailpoet , 

Does MailPoet provide any mechanism (like a PHP filter) for overriding the default sender for specific emails? 

WooCommerce is another plugin that allows users to specify the "From" name & email address for their emails. I tested out an order with WooCommerce and it's the same situation there — MailPoet overrides WooCommerce's "From" name & email address settings. 

I think for any site running WooCommerce, they would expect the "From" name & email address specified for WooCommerce emails to be the one that's used — not the one set for MailPoet. 

I hope there's a way to resolve this. Otherwise we end up having to tell customers _not_ to use the MailPoet sending method for all their emails, which is a shame for the customer and obviously also reduces the number of emails getting sent through your service.

**Comment by pavel-mailpoet** [comment]
Hello @ericnicolaas ,
we don't provide any mechanism for that. Could you please elaborate on what is the use-case? Why cannot users keep the same sender email address in both plugins?

**Comment by ericnicolaas** [comment]
Hi @pavel-mailpoet, 

Sure, so for the particular user who ran into this issue and let us know about it (a mutual customer of ours), they wanted Charitable's emails to be sent from a `donate@example.org`, whereas their MailPoet emails should be sent from `admin@example.org`. 

Similarly, if a site is running WooCommerce and using MailPoet for their newsletters, I could imagine some sites would like the ability to send their WooCommerce email from a store-specific email address (say `store@example.com`) and maybe their newsletter would go from `newsletter@example.com`. 

Ultimately, it matters because whoever is checking the email address used for MailPoet may not be the same person responsible for donations (in Charitable's case), or for purchases (in the WooCommerce scenario).

**Comment by ericnicolaas** [comment]
I haven't dug deep into this how your mailer classes work, but one option that I think could be simple enough is to have a boolean filter like `mailpoet_use_mailpoet_sender_details` or something like that. 

Then all a plugin like ours has to do is set that to `false` before we call `wp_mail()`. 

```
add_filter( 'mailpoet_use_mailpoet_sender_details', '__return_false' );

wp_mail( ... );
```

**Comment by NeosinneR** [comment]
Hi @ericnicolaas,

I understand the need for this and it indeed makes sense. Unfortunately, due to our current priorities, we won't be able to implement anything that would help you anytime soon.

What you proposed would work, but then emails might end up in the spam (not unusual when sending with your host) and the customer would blame us because he specifically set all emails to be sent through MailPoet and now they would end up in spam. Not a situation we want to be in.

There are other ways of allowing different from address, but none is easy enough to do it in between our current priorities. I created a ticket in our internal system to allow this in the future, but I can't give you any ETA.

**Comment by ericnicolaas** [comment]
Thanks for the feedback @NeosinneR. Would it be possible to have this issue updated when some progress has made with this? For now, I will just advise customers to only use MailPoet's sending service for MailPoet newsletters (i.e. don't use it for transactional emails).

**Comment by NeosinneR** [comment]
@ericnicolaas will do. Added a note to our ticket to notify you once it's done ;-)

**Comment by sdenengelsman** [comment]
Is there any update on this?

I'm having the same issue and to be honest, I think this is quite a big issue, even though it is by design.

Example:
We use a number of plugins and there are quite a few events on our website that require an email notification.
Pending posts? Those notifications go to the site admins and should be sent from do-not-reply@our-domain.com.
The Mailpoet newsletters should go from newsletters@our-domain.com.
Payment info should go from payments@our-domain.com.
Etcetera... the list goes on.

It would look silly and be incredibly unpractical if all emails, no matter what they are about, come from newsletters@our-domain.com.

Is there at least an option to only send the newsletters through the Mailpoet sending service?

**Comment by arcadreams** [comment]
Hello,

I am also interested because without this functionality I am considering stopping using Mailpoet…

Thanks,
Sergio

**Comment by ericnicolaas** [comment]
@NeosinneR and @pavel-mailpoet: Did any progress ever get made on this? Obviously from the others responding here this is something others need as well. This issue was closed and then acknowledged as something needed without being re-opened, so it's hard to see if there is any progress towards this.

**Comment by NeosinneR** [comment]
@ericnicolaas @arcadreams unfortunately there is no progress on this. We gather feedback from our users on [feedback.mailpoet.com](https://feedback.mailpoet.com/feature-requests?sort=top) and while I understand this would help you a lot, it's a feature not that many users [requested](https://feedback.mailpoet.com/feature-requests/p/multiple-default-emails). Please, add a vote to the request to increase the chance we'll implement it in the future.

**Comment by ericnicolaas** [comment]
Thanks @NeosinneR. I've added my vote to that feature request. I'll note though that the same change would also help with addressing all of the following feature requests:

- https://feedback.mailpoet.com/feature-requests/p/option-to-respect-the-original-reply-to-when-smtp-service-is-used-for-all-wordpr
- https://feedback.mailpoet.com/feature-requests/p/optional-default-transactional-email-send-address
- https://feedback.mailpoet.com/feature-requests/p/use-separate-email-address-for-subscription-confirmations

There may be more but those are what I found in a quick look through the feedback requests. Clearly the ability to change the Reply-To/From email address could be helpful in a variety of ways.
